staging/src/k8s.io/apiserver/pkg/server/options/etcd.go

```diff
@@ -346,6 +353,14 @@ func ParseWatchCacheSizes(cacheSizes []string) (map[schema.GroupResource]int, er
 	return watchCacheSizes, nil
 }
 
+// ParseWatchCacheSizesWithOnce is a wrapper of parseWatchCacheSizes for cleanliness and testing.
+func ParseWatchCacheSizesWithOnce(cacheSizes []string) (map[schema.GroupResource]int, error) {
+	parseWatchCacheSizesOnce.Do(func() {
+		watchCacheSizes, parseWatchCacheSizesErr = parseWatchCacheSizes(cacheSizes)
+	})
+	return watchCacheSizes, parseWatchCacheSizesErr
```

Comment:
Well...this argument is specified by `--watch-cache-sizes` for now, actually all callers will use this same value, and further more we should make it as a settled value, because once user specified a value, it should not be modified, it's more semantically using the same argument.